cpp/src/arrow/util/bit-util.h

```diff
@@ -686,6 +690,65 @@ void GenerateBitsUnrolled(uint8_t* bitmap, int64_t start_offset, int64_t length,
   }
 }
 
+// A function that visits each bit in a bitmap and calls a visitor function with a
+// boolean representation of that bit. This is intended to be analogous to
+// GenerateBits.
+template <class Visitor>
+void VisitBits(const uint8_t* bitmap, int64_t start_offset, int64_t length,
+               Visitor&& visit) {
+  for (int64_t index = start_offset; index < start_offset + length; ++index) {
+    visit(arrow::BitUtil::GetBit(bitmap, index));
+  }
+}
+
+// Like VisitBits(), but unrolls its main loop for better performance.
+template <class Visitor>
+void VisitBitsUnrolled(const uint8_t* bitmap, int64_t start_offset, int64_t length,
+                       Visitor&& visit) {
+  // Avoid having to include "BitUtil::" throughout this function.
```

Comment:
I don't think you need a comment to explain this :-)

Reply:
Fixed